types/vote.go

```diff
@@ -389,3 +400,40 @@ func VotesToProto(votes []*Vote) []*tmproto.Vote {
 	}
 	return res
 }
+
+func SignAndCheckVote(
+	vote *Vote,
+	privVal PrivValidator,
+	chainID string,
+	extensionsEnabled bool,
+) (bool, error) {
+	v := vote.ToProto()
+	if err := privVal.SignVote(chainID, v); err != nil {
```

Comment:
In what cases is failing to sign a vote recoverable?

Reply:
That's a good question.
I added the recoverable flag because this is part of a small refactoring, and didn't want to modify the original logic in any way... in the sense I didn't want to add a panic where there used to be none.
If you look at the changes of this PR in the only callsite in production code (`consensus/state.go` line 2324) you will see that there was no panic when `SignVote` failed, so I wanted to keep that (battle-tested) behavior unchanged. However, the new checks (introduced by `SignAndCheckVote`, which involve vote extensions are situations we can't (or don't want to) recover from.

I'll see if I can add a meaningful comment.